src/model/selection.js

```diff
@@ -322,36 +326,66 @@ export default class Selection {
 	 *		const position = new Position( root, path );
 	 *		selection.setTo( position );
 	 *
-	 * 		// Sets range at the position of given element and optional offset.
+	 * 		// Sets range at the position of given node and offset.
 	 *		const paragraph = writer.createElement( 'paragraph' );
 	 *		selection.setTo( paragraph, offset );
 	 *
+	 *		// Sets range inside the node.
+	 *		const paragraph = writer.createElement( 'paragraph' );
+	 *		selection.setTo( paragraph, 'in' );
+	 *
+	 *		// Sets range on the node.
+	 *		const paragraph = writer.createElement( 'paragraph' );
+	 *		selection.setTo( paragraph, 'on' );
+	 *
 	 * 		// Removes all ranges.
 	 *		selection.setTo( null );
 	 *
 	 * @param {module:engine/model/selection~Selection|module:engine/model/documentselection~DocumentSelection|
-	 * module:engine/model/position~Position|module:engine/model/element~Element|
+	 * module:engine/model/position~Position|module:engine/model/node~Node|
 	 * Iterable.<module:engine/model/range~Range>|module:engine/model/range~Range|null} selectable
-	 * @param {Boolean|Number|'before'|'end'|'after'} [backwardSelectionOrOffset]
+	 * @param {Object|Number|'before'|'end'|'after'|'on'|'in'} [optionsOrPlaceOrOffset]
+	 * @param {Boolean} [optionsOrPlaceOrOffset.backward]
+	 * @param {Object} [options]
+	 * @param {Boolean} [options.backward]
```

Comment:
We should documentation simple:
```
* @param {Object|Number|'before'|'end'|'after'|'on'|'in'} [placeOrOffset]
* @param {Object} [options]
* @param {Boolean} [options.backward]
```

Since the second parameter is optional it is clear that the third parameter can the second parameter too. Even if it is not clear, there are examples in docs.

Reply:
Also, you can add some descriptions to these parameters.

Reply:
I've added them to the view, probably forget to add it here too. I'll fix this in model & view selections.

Reply:
Should the second parameter allow 'Object'? As I understand, it should be `{Number|'before'|'end'|'after'|'on'|'in'} [placeOrOffset]` now.

Reply:
Yep. The information that it is optional should be enough. If it is `Object` it means there is no `placeOrOffset` and the parameter is `options`. We can not do this in the code but we can do it in docs.

Reply:
Hmm, I've checked it and TS throws an error when two parameters are optional and only the second is passed as first (Only valid invocation is `fn( undefined, secondArgument )`. Maybe that's why I intuitively didn't like this idea. Hope we won't have problems with it as there was an idea to generate some TS declaration files in the future.